test/server.node.js

```diff
@@ -0,0 +1,79 @@
+/* eslint-env mocha */
+
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+const sinon = require('sinon')
+const pWaitFor = require('p-wait-for')
+
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const multiaddr = require('multiaddr')
+
+const pipe = require('it-pipe')
+
+const IDJSON = require('./fixtures/peer-server.json')
+const Server = require('../src/server')
+const Stardust = require('../src')
+
+const { createPeer } = require('./utils')
+const mockUpgrader = {
+  upgradeInbound: maConn => maConn,
+  upgradeOutbound: maConn => maConn
+}
+const SERVER_URL = multiaddr('/ip4/127.0.0.1/tcp/5893/ws/p2p-stardust/p2p/QmY7t45zkBxzw5rtWQp3oAjzoJGTjh49f7LNz4YPBtTpqy')
+
+describe('server', () => {
+  let server
+
+  beforeEach(async () => {
+    const peerId = await PeerId.createFromJSON(IDJSON)
+    const peerInfo = await PeerInfo.create(peerId)
+    const addresses = [multiaddr('/ip6/::/tcp/5893/ws')]
+
+    server = new Server({ addresses, peerInfo })
+    await server.start()
+  })
+
+  afterEach(async () => {
+    sinon.reset()
```

Comment:
Do you want `sinon.restore()`? I believe reset will keep the fakes in place but just reset them. `restore` will remove all fakes and restore the originals.